cmd/darvaza/serve.go

```diff
@@ -10,12 +10,35 @@ import (
 	"github.com/spf13/cobra"
 )
 
+// pr is an interface which is implemented by all proxies
+type pr interface {
```

Comment:
I tend to prefer having these interfaces public and living on a dedicated zero-dependency namespace. e.g. `types/`